src/engine/IndexScan.cpp

```diff
@@ -266,43 +266,43 @@ ScanSpecificationAsTripleComponent IndexScan::getScanSpecification() const {
 
 // ___________________________________________________________________________
 Permutation::IdTableGenerator IndexScan::getLazyScan(
-    const IndexScan& s, std::vector<CompressedBlockMetadata> blocks) {
-  const IndexImpl& index = s.getIndex().getImpl();
+    std::vector<CompressedBlockMetadata> blocks) const {
+  const IndexImpl& index = getIndex().getImpl();
   std::optional<Id> col0Id;
-  if (s.numVariables_ < 3) {
-    col0Id = s.getPermutedTriple()[0]->toValueId(index.getVocab()).value();
+  if (numVariables_ < 3) {
+    col0Id = getPermutedTriple()[0]->toValueId(index.getVocab()).value();
   }
   std::optional<Id> col1Id;
-  if (s.numVariables_ < 2) {
-    col1Id = s.getPermutedTriple()[1]->toValueId(index.getVocab()).value();
+  if (numVariables_ < 2) {
+    col1Id = getPermutedTriple()[1]->toValueId(index.getVocab()).value();
   }
```

Comment:
I think this can be refactored to use the new `getScanSpecification` facilities.
Then we can also drop the requirement to have at least one variable
(see [here](https://qlever.cs.uni-freiburg.de/wikidata/B2oXvl).